File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_e2/organic_string_1_valid_parentheses.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public boolean isValid(String s) {
        StringBuilder stack_str = new StringBuilder();
        Map<Character, Character> mapping = new HashMap<>();
        mapping.put(')', '(');
        mapping.put('}', '{');
        mapping.put(']', '[');
        
        for (char c : s.toCharArray()) {
            if (!mapping.containsKey(c)) {
                stack_str.append(c);
            } else {
                if (stack_str.length() == 0) {
                    return false;
                }
                if (stack_str.charAt(stack_str.length() - 1) != mapping.get(c)) {
                    return false;
                }
                stack_str.deleteCharAt(stack_str.length() - 1);
            }
        }
        return stack_str.length() == 0;
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_e2/organic_string_1_valid_parentheses.py
def isValid(s):
    stack_str = ""
    mapping = {")": "(", "}": "{", "]": "["}
    
    for c in s:
        if c not in mapping:
            stack_str += c
        else:
            if len(stack_str) == 0:
                return False
            if stack_str[-1] != mapping[c]:
                return False
            stack_str = stack_str[:-1]
            
    return len(stack_str) == 0
